email_notification/validators.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+from django.core.exceptions import ValidationError
+from sentry_sdk import capture_message
+
+from reservations.models import Reservation
+
+MAX_HTML_FILE_SIZE = 150000
+
+
+def html_file_validator(value):
```

Comment:
This just means that in the way we are validating two different things: uploaded file and text content of the email. But I guess that is fine.

I started to wonder one thing. The ticket said that we should run the HTML content through "email rendering", but what we **really** need is to run it though `EmailTemplateValidator.validate_string()`. Internally, that is what rendering does anyway and that is what we care about. In the end, we don't care if content is valid HTML - we just care about the tags.

With that change testing becomes much easier because we don't have to have an actual `Reservation` for the context.

Reply:
So, I changed the validation. I removed it from the model level and added it to `admin.py`. That way I could easily extended and use the existing validation. I also removed the need for actual Reservation. Now it just calls `validate_string` and that is it!

Better? Worse? 😄

Reply:
hmph, I think I saw the change and already commented it in other thread 😂